sources/Context.cpp

```diff
@@ -163,162 +201,194 @@ Context::areInMainConnectedComponent(const std::vector<boost::shared_ptr<DYN::Bu
   return result;
```

Comment:
I think the expression `result = result | nvalid` is wrong/confusing. The operator` | `is a bitwise operator, we should use a logical (boolean) operator, ` || `. Also, the condition does not follow what the name of the function states. The name suggests that a value of true would be returned if all the buses are in the main connected component, thus we should use ` && `. 

Anyway, I think that to check if an element is in the main connected component, it would be enough to check if just one of its end buses is in the main connected component (and there is no need to check the rest) ... we will rewrite and rename the function to reflect this.